### PR TITLE
Add multi error exceptions

### PR DESCRIPTION
**Pull request by denniskoster**
**Changes**
- Changed type of `$status` in exceptions to integer
- Added a couple of exceptions that extend the `AbstractMultiErrorException`

### CODE REVIEW

File: src/Exceptions/InvalidCredentialsException.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * reflecting invalid credentials.
+ */
+class InvalidCredentialsException extends AbstractMultiErrorException
+{
+    /**
+     * @param array          $errors
+     * @param int            $status
+     * @param Throwable|null $previous
+     */
+    public function __construct(array $errors, int $status = 403, ?Throwable $previous = null)
```
[the remark this answers is not in the document]
Comment (denniskoster) [reply src/Exceptions/InvalidCredentialsException.php]: According to Wikipedia:
```
Similar to 403 Forbidden, but specifically for use when authentication is required and has failed or has not yet been provided. The response must include a WWW-Authenticate header field containing a challenge applicable to the requested resource.
```

As far as I know we don't send a www-authenticate header, so 403 seemed applicable to me.

[the remark this answers is not in the document]
Comment (NickVries) [reply src/Exceptions/InvalidCredentialsException.php]: Also according to Wikipedia:

```
HTTP 403 is returned when the client is not permitted access to the resource for some reason besides authentication. 
```

I'm not an expert on this, but this doesn't really apply here right?

[the remark this answers is not in the document]
Comment (denniskoster) [reply src/Exceptions/InvalidCredentialsException.php]: Not sure where you found that, but it says the following for me:
```
The request was valid, but the server is refusing action. The user might not have the necessary permissions for a resource, or may need an account of some sort.
```

[the remark this answers is not in the document]
Comment (denniskoster) [reply src/Exceptions/InvalidCredentialsException.php]: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_errors

**Pull request merged**